pkg/cli/clients/management.go

```diff
@@ -64,138 +69,148 @@ var (
 	}
 )
 
-// ListAllResourcesByType lists the all the resources within a scope
-//
-
-// ListAllResourcesByType retrieves a list of all resources of a given type from the root
-// scope, and returns them in a slice of GenericResource objects, or an error if one occurs.
-func (amc *UCPApplicationsManagementClient) ListAllResourcesByType(ctx context.Context, resourceType string) ([]generated.GenericResource, error) {
-	results := []generated.GenericResource{}
-
-	client, err := generated.NewGenericResourcesClient(amc.RootScope, resourceType, &aztoken.AnonymousCredential{}, amc.ClientOptions)
+// ListResourcesOfType lists all resources of a given type in the configured scope.
+func (amc *UCPApplicationsManagementClient) ListResourcesOfType(ctx context.Context, resourceType string) ([]generated.GenericResource, error) {
+	client, err := amc.createGenericClient(amc.RootScope, resourceType)
 	if err != nil {
-		return results, err
+		return nil, err
 	}
 
+	results := []generated.GenericResource{}
 	pager := client.NewListByRootScopePager(&generated.GenericResourcesClientListByRootScopeOptions{})
 	for pager.More() {
-		nextPage, err := pager.NextPage(ctx)
+		page, err := pager.NextPage(ctx)
 		if err != nil {
-			return results, err
+			return nil, err
 		}
-		applicationList := nextPage.GenericResourcesList.Value
-		for _, application := range applicationList {
-			results = append(results, *application)
+
+		for _, resource := range page.Value {
+			results = append(results, *resource)
 		}
 	}
```

Comment:
Were we doing this in unnecessarily complicated way before? I am not sure why we changed this to page.Value from page.GenericResourcesList.Value

Reply:
It's the same. `page.Value` and `page.GenericResourcesList.Value` are actually the same field.

What's happening here is that the `GenericResourcesList` is embedded. 


```go
type GenericResourcesListResponse struct {
    GenericResourcesList
}

type GenericResourcesList struct {
    Value []GenericResource
}
```

Reply:
When two things are equivalent, prefer less code 😆